Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/EditorModeOutline.azsl

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <EditorModeCommon.azsli>
+#include <EditorModeDepthTransition.azsli>
+
+partial ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
+{    
+    float m_lineThickness;
+    float4 m_lineColor;
+}
+
+//! Applies an outline effect to the entity mask of the specified line thickness.
+//! @note There are known issues with pimples due to z-fighting of coincident surfaces in the entity mask that will be addressed it LYN-10304.
+float selectedEntityOutliner(const float2 uv, const float lineThickness)
+{
+    const float2 maskSize = PassSrg::GetImageSize(PassSrg::m_entityMask);
+    const float2 unormalizedUv = float2(uv.x * maskSize.x, uv.y * maskSize.y);
+    const float2 maskSizeReciprocal = float2(1., 1.0) / maskSize;
+    float r = 0.;
+
+    // Selected entity mask value
+    float S = PassSrg::m_entityMask.Sample(PassSrg::PointSampler, unormalizedUv * maskSizeReciprocal).a;
+
+    if(S > 0.0)
+    {
+        return 0.0;
+    }
+    
+    // Outliner kernal
+    for(float u = -lineThickness; u <= lineThickness; u+=1.)
+    {
+        for(float v = -lineThickness; v <= lineThickness; v+=1.)
+        {
+            r += PassSrg::m_entityMask.Sample(PassSrg::PointSampler, (unormalizedUv + float2(u,v)) * maskSizeReciprocal).r;
+        }
+    }
+    
+    // Only apply the outliner effect to fragments not part of S
+    return min(1.0, (r / lineThickness));
```

Comment:
/ (lineThickness * lineThickness)
Will probably give you a better falloff (with how it currently is most pixels in the outline are going to have a full outline amount of 1 because you are sampling a neighborhood of (3 + 1 + 3)^2 = 49 pixels and then dividing by 3 (if that's what you want though disregard this comment)

Reply:
You are correct, it should've been the number of elements in the kernel matrix 👍